src/server/req_quejob.c

```diff
@@ -2199,9 +2224,10 @@ req_resvSub(struct batch_request *preq)
 	 * but the structure field would be an addition to the
 	 * "quick save" area of the server - can't do
 	 */
-
-	(void)snprintf(qbuf, sizeof(qbuf), "%c%lld", PBS_RESV_ID_CHAR,
-			next_svr_sequence_id);
+	if (generate_objid(qbuf, NULL, 2, PBS_RESV_ID_CHAR) != 0) {
```

Comment:
This generates a queue name for the reservation, so just do a strchr on 'rid' and dup the portion before the '.' instead of adding the big if (!clusterid) blocks inside generate_objid().